src/TraceEvent/Computers/TraceProcess.cs

```diff
@@ -27,24 +27,27 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceProcessesExtensions
     {
+        internal static readonly string ProcessesKey = "Computers/Processes";
+
         public static void NeedProcesses(this TraceEventDispatcher source)
         {
             TraceProcesses processes = source.Processes();
-            if (processes == null || m_currentSource != source)
+            if (processes == null)
             {
```

Comment:
That is not true.  Please bear this out with your concurrent test suite?  The previous check is a loose guarantee, but it is required.